cng/aes.go

```diff
@@ -119,34 +119,45 @@ func (c *aesCipher) NewGCMTLS() (cipher.AEAD, error) {
 	return newGCM(c.key, true)
 }
 
-type aesCBC struct {
-	kh      bcrypt.KEY_HANDLE
-	iv      [aesBlockSize]byte
-	encrypt bool
+type cbcCipher struct {
+	kh bcrypt.KEY_HANDLE
+	// us aesBlockSize as the max block size for all ciphers
+	// to avoid allocations.
+	iv        [aesBlockSize]byte
```

Comment:
Typo, and I think the comment could be a little clearer:

```suggestion
	// Use aesBlockSize, the max of all supported cipher block sizes.
	// The array avoids allocations (vs. a slice).
	iv        [aesBlockSize]byte
```

(Fun that once we're post-1.21, we could use `[max(aesBlockSize, desBlockSize)]byte` here.)